src/cipher/testsuite/testsuite.go

```diff
@@ -0,0 +1,264 @@
+package testsuite
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/skycoin/skycoin/src/cipher"
+	secp256k1 "github.com/skycoin/skycoin/src/cipher/secp256k1-go"
+)
+
+// InputTestDataJSON contains hashes to be signed
+type InputTestDataJSON struct {
+	Hashes []string `json:"hashes"`
+}
+
+// KeysTestDataJSON contains address, public key,  secret key and list of signatures
+type KeysTestDataJSON struct {
+	Address    string   `json:"address"`
+	Secret     string   `json:"secret"`
+	Public     string   `json:"public"`
+	Signatures []string `json:"signatures,omitempty"`
+}
+
+// SeedTestDataJSON contains data generated by Seed
+type SeedTestDataJSON struct {
+	Seed string             `json:"seed"`
+	Keys []KeysTestDataJSON `json:"keys"`
+}
+
+// InputTestData contains hashes to be signed
+type InputTestData struct {
+	Hashes []cipher.SHA256
+}
+
+// ToJSON converts InputTestData to InputTestDataJSON
+func (d *InputTestData) ToJSON() *InputTestDataJSON {
+	hashes := make([]string, len(d.Hashes))
+	for i, h := range d.Hashes {
+		hashes[i] = h.Hex()
+	}
+
+	return &InputTestDataJSON{
+		Hashes: hashes,
+	}
+}
+
+// InputTestDataFromJSON converts InputTestDataJSON to InputTestData
+func InputTestDataFromJSON(d *InputTestDataJSON) (*InputTestData, error) {
+	hashes := make([]cipher.SHA256, len(d.Hashes))
+	for i, h := range d.Hashes {
+		var err error
+		hashes[i], err = cipher.SHA256FromHex(h)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &InputTestData{
+		Hashes: hashes,
+	}, nil
+}
+
+// KeysTestData contains address, public key,  secret key and list of signatures
+type KeysTestData struct {
+	Address    cipher.Address
+	Secret     cipher.SecKey
+	Public     cipher.PubKey
+	Signatures []cipher.Sig
+}
+
+// ToJSON converts KeysTestData to KeysTestDataJSON
+func (k *KeysTestData) ToJSON() *KeysTestDataJSON {
+	sigs := make([]string, len(k.Signatures))
+	for i, s := range k.Signatures {
+		sigs[i] = s.Hex()
+	}
+
+	return &KeysTestDataJSON{
+		Address:    k.Address.String(),
+		Secret:     k.Secret.Hex(),
+		Public:     k.Public.Hex(),
+		Signatures: sigs,
+	}
+}
+
+// KeysTestDataFromJSON converts KeysTestDataJSON to KeysTestData
+func KeysTestDataFromJSON(d *KeysTestDataJSON) (*KeysTestData, error) {
+	addr, err := cipher.DecodeBase58Address(d.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	s, err := cipher.SecKeyFromHex(d.Secret)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := cipher.PubKeyFromHex(d.Public)
+	if err != nil {
+		return nil, err
+	}
+
+	var sigs []cipher.Sig
+	if d.Signatures != nil {
```

Comment:
Seems no need to check if `d.Signatures` is nil, `range` would take care of it.

Reply:
I want it to be exact copy, so that if d.Signatures is nil, new sigs will be nil, but if d.Signatures is non-nil empty array, new sigs will be non-nil empty array